extensions/amp-story/0.1/amp-story-auto-ads.js

```diff
@@ -89,8 +90,25 @@ export class AmpStoryAutoAds extends AMP.BaseElement {
 
   /** @override */
   layoutCallback() {
-    this.schedulePage_();
-    return Promise.resolve();
+    return this.createStoryLoadPromise_()
+        .then(() => {
+          this.readConfig_();
+          this.schedulePage_();
+        });
+  }
+
+
+  /**
+   * promise that resolves on STORY_LOADED event
+   * @private
+   * @return {!Promise}
+   */
+  createStoryLoadPromise_() {
+    return new Promise(resolve => {
+      this.ampStory_.element.addEventListener(EventType.STORY_LOADED, () => {
```

Comment:
`this.ampStory_.element.addEventListener(EventType.STORY_LOADED, resolve)`

Reply:
ah of course 😎. done.